protected/humhub/modules/content/components/ContentContainerActiveRecord.php

```diff
@@ -341,4 +365,86 @@ public function isVisibleFor($visibility)
         return $this->visibility == $visibility;
     }
 
+    /**
+     * Checks if the Content Container has Tags
+     *
+     * @return boolean has tags set
+     */
+    public function hasTags()
+    {
+        return count($this->getTags()) > 0;
+    }
+
+    /**
+     * Returns an array with related Tags
+     *
+     * @return array
+     */
+    public function getTags()
+    {
+        return ContentContainerTag::find()
+            ->select('name')
+            ->leftJoin('contentcontainer_tag_relation', 'id = tag_id')
+            ->where(['contentcontainer_id' => $this->contentcontainer_id])
+            ->andWhere(['contentcontainer_class' => get_class($this)])
+            ->column();
+    }
+
+    /**
+     * Update related tags
+     */
+    protected function updateTags()
+    {
+        if (!isset($this->updatedTags)) {
```

Comment:
@buddh4

> Don't we have a form model available for this?

No, I don't find. Only `humhub\modules\space\models\Space` is used for all forms.

> Also did you test this when removing all tags?

Yes, I did. I implemented only removing the relations, but keep old tags in DB if even they are not used by any other Container, should we do this?

> Could we move this whole tag logic to the `ContentContainerTag` class instead.

Ok, I have moved but only to the `ContentContainerTagRelation`, I think it is more related, because the methods remove the relations and not tags. Commit https://github.com/humhub/humhub/pull/4867/commits/364d66a97dab7b2a9c9b29e1937b8f58674335f1.